internal/setup/setup.go

```diff
@@ -418,14 +423,17 @@ func parseRelease(baseDir, filePath string, data []byte) (*Release, error) {
 		return nil, fmt.Errorf("%s: no archives defined", fileName)
 	}
 
-	keyringsByName := make(map[string]openpgp.KeyRing, 0)
-
-	for keyringName, keyringArmored := range yamlVar.Keyrings {
-		keyring, err := openpgp.ReadArmoredKeyRing(strings.NewReader(keyringArmored))
+	// Decode the public keys and match against provided IDs.
+	pubKeys := make(map[string]*packet.PublicKey, len(yamlVar.PublicKeys))
+	for keyName, yamlPubKey := range yamlVar.PublicKeys {
+		key, err := DecodePublicKey([]byte(yamlPubKey.Armor))
 		if err != nil {
-			return nil, fmt.Errorf("%s: cannot parse keyring %q: %w", fileName, keyringName, err)
+			return nil, fmt.Errorf("%s: invalid public key %q: %w", fileName, keyName, err)
```

Comment:
Previous was worded per our conventions, new one is not and is infering something it probably should not.

Instead, `"cannot decode public key %q: %w"`.